### DISCUSSION TITLE
인프라 설계에 대해서 논의하고 싶습니다.

### DISCUSSION THREAD

**Discussion by 02ggang9** [BDD-CLUB/01-doo-re-back#49] [논의 사항]
RDS를 도입하는 과정에서 몇 가지 문제에 직면했습니다. 

### RDS 접근 제어
기본적으로 데이터베이스는 외부에서 접근이 가능하면 안됩니다. 오직 ec2 인스턴스만 RDS에 접근이 가능하게 하려면 VPC, Private Subnet, 라우팅 테이블, 인터넷 게이트웨이 설정 등을 해야 합니다. 도전해보고 싶은 과제이긴 하나 추가적인 비용이 발생하므로 저희가 선택할 수는 없을 것 같습니다. 또, 초기 dev ec2 인스턴스를 설정할 때 subnet 설정을 못 해 인스턴스를 새로 파야합니다. 지금까지 했던 모든 과정을 다시 해야 하는 문제가 생깁니다.

### RDS 외부 접근 허용
외부에서 RDS 인스턴스에 접근을 허용한다면 보안적으로 취약해집니다. 하지만 추가적인 비용 문제와 수고로움을 덜 수 있습니다. 또, RDS 인스턴스를 사용할 수 있게 되므로 자동 백업 및 스토리지 확보도 가능합니다.

### 오직 EC2 인스턴스로만
기존 방식대로 EC2 인스턴스에 모든 컨테이너를 띄운다면 RDS에서 제공하는 자동 백업 기능을 사용할 수 없기 때문에 추가적인 설정을 해야 합니다. 또, RDS에서 제공하는 스토리지를 EC2에서 부담해야 합니다.

### 총 정리
1. RDS + 외부 접근 x
보안적으로 우수하고 실무와 가깝다고 생각합니다. 하지만 추가적인 비용 발생과 EC2 인스턴스를 재 생성해야 하는 수고로움, 추가적인 학습 비용이 발생합니다.

2. RDS + 외부 접근 o
RDS에서 제공하는 자동 백업 기능과 스토리지 확보의 이점이 있습니다. 하지만 외부에서 접근이 가능하다는 점이 보안적으로 취약하다고 생각합니다. 또, 약간의 배포 코드 수정이 필요합니다.

3. 기존 EC2 인스턴스처럼
DB 자동 백업 설정만 해 주면 됩니다. 하지만 RDS에서 제공하는 자동 백업 기능 포기와 스토리지가 축소된다는 단점이 있습니다.

금요일날 dev, prod 인프라 설게에 대한 얘기를 마치고 싶습니다!! 제가 잘못 이해하고 있거나 또 다른 방법이 있다면 알려주시면 감사드립니다~

긴 글 읽어주셔서 감사합니다

**Comment by JJimini** [comment]
1번으로 결정이 되었으나 추가 비용이 실제로 드는가에 대해서는 공부 필요
